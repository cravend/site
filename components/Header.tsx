import React, { FC } from "react";
import { Box, Image, Heading, Container } from "theme-ui";
import useLocalization from "../src/hooks/useLocalization";
import Hero from "./Hero";

const Header: FC = () => {
  const [t] = useLocalization();

  return (
    <Hero isDiagonal>
      <Container
        variant="copy"
        m="0 auto"
        py={4}
        sx={{
          display: "flex",
          height: "100%",
          flexDirection: ["column", "row"],
          alignItems: "center",
          justifyContent: "space-around",
        }}
      >
        <Image
          src="/images/dalton-craven.jpg"
          alt={t("pfp")}
          width={150}
          sx={{ borderRadius: "50%" }}
        />
        <Box
          ml={[0, 20]}
          p={1}
          sx={{
            marginLeft: [0, 2],
            marginTop: [2, 0],
            textAlign: ["center", "right"],
            color: "white",
          }}
        >
          <Heading as="h1" variant="title">
            {t("name")}
          </Heading>
          <Heading
            as="h3"
            variant="subtitle"
            sx={{ textTransform: "lowercase" }}
          >
            {t("tagline")}
          </Heading>
        </Box>
      </Container>
    </Hero>
  );
};

export default Header;
