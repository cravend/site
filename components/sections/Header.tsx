import { useTranslations } from "next-intl";
import Link from "next-intl/link";

function Header() {
  const t = useTranslations("header");

  return (
    <header>
      <h1>{t("intro.title")}</h1>
      <p>{t("intro.subtitle")}</p>
      <nav>
        <p>{t("nav.intro")}</p>
        <ul>
          <li>
            <Link href="/" locale="en">
              {t("nav.options.en")}
            </Link>
          </li>
          <li>
            <Link href="/" locale="fr">
              {t("nav.options.fr")}
            </Link>
          </li>
          <li>
            <Link href="/" locale="ar">
              {t.rich("nav.options.ar")}
            </Link>
          </li>
        </ul>
      </nav>
    </header>
  );
}

export default Header;
