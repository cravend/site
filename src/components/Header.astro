---
import { t, l } from "astro-i18n";
---

<header>
  <h1>{t("header.intro.title")}</h1>
  <p>{t("header.intro.subtitle")}</p>
  <nav>
    <p>{t("header.nav.intro")}</p>
    <ul>
      <li>
        <a href={l("/", {}, "en")}>{t("header.nav.options.en")}</a>
      </li>
      <li>
        <a href={l("/", {}, "fr")}>{t("header.nav.options.fr")}</a>
      </li>
      <li>
        <a href={l("/", {}, "ar")} set:html={t("header.nav.options.ar")} />
      </li>
    </ul>
  </nav>
</header>
