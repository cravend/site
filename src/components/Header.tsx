import Image from "next/image";
import useLocalization from "../i18n/useLocalization";
import styles from "../styles/modules/Header.module.scss";
import profilePic from "../assets/images/dalton-craven.webp";
import Hero from "./Hero";
import type { FC } from "react";

const Header: FC = () => {
  const [t] = useLocalization();

  return (
    <Hero isDiagonal>
      <div className={styles.container}>
        <div className={styles.profile}>
          <Image className={styles.pfp} src={profilePic} alt={t("pfp")} />
        </div>
        <div className={styles.text_container}>
          <h1 className={styles.title}>{t("name")}</h1>
          <h3 className={styles.subtitle}>{t("tagline")}</h3>
        </div>
      </div>
    </Hero>
  );
};

export default Header;
