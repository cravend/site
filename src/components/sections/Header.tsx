import Image from "next/image";
import { useTranslations } from "next-intl";

import profilePic from "../../assets/images/dalton-craven.webp";
import styles from "../../styles/modules/sections/Header.module.scss";
import Hero from "../Hero";

const Header = () => {
  const t = useTranslations("sections.Header");

  return (
    <Hero isDiagonal>
      <div className={styles.container}>
        <div className={styles.profile}>
          <Image
            className={styles.pfp}
            src={profilePic}
            alt={t("pfp")}
            width={150}
            height={150}
          />
        </div>
        <div className={styles.text_container}>
          <h1 className={styles.title}>{t.rich("name")}</h1>
          <p className={styles.subtitle}>
            <b>{t.rich("tagline")}</b>
          </p>
        </div>
      </div>
    </Hero>
  );
};

export default Header;
