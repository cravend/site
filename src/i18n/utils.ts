import { locales, defaultLocale, RTL_LOCALES } from "./config";

import type { Locale } from "./config";
import type { NextRouter } from "next/router";

/**
 * @description This function is used to check if a string is a valid locale.
 *
 * @param localeToTest The string to check.
 * @returns True if the string is a valid locale, false otherwise.
 */
export const isLocale = (localeToTest?: string): localeToTest is Locale =>
  locales.some((locale) => locale === localeToTest);

/**
 * @description This function is used to get the current locale.
 *
 * If a custom locale is set by \@next/router, it will be returned.
 * If no custom locale is set, the router's default locale will be returned.
 * If the router's default locale isn't set, the default locale will be returned.
 *
 * @param router The NextRouter instance.
 * @returns The current locale.
 */
export const getLocale = (router: NextRouter | null) => {
  if (!router) return defaultLocale;
  if (isLocale(router.locale)) return router.locale;
  if (isLocale(router.defaultLocale)) return router.defaultLocale;
  return defaultLocale;
};

/**
 * @description This function is used to toggle the current locale.
 *
 * @param locale The current locale to be toggled.
 * @returns The new locale.
 */
export const toggleLocale = (locale: Locale) =>
  locale === locales[0] ? locales[1] : locales[0];

/**
 * @description This function is used to get the text direction of the current locale.
 *
 * @param locale The locale to be checked
 * @returns True if the locale is read right-to-left, false otherwise
 */
export const isRtl = (locale: Locale) => RTL_LOCALES.includes(locale);
