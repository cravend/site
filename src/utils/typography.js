import Typography from 'typography'

const typography = new Typography({
  baseLineHeight: 1.666,
  headerFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  bodyFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  googleFonts: [
    {
      name: 'Poppins',
      styles: ['300', '600', '700']
    }
  ]
})

export default typography
