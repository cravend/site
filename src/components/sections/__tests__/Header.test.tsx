import { render } from "../../../tests/utils";
import Header from "../Header";

describe("<Header />", () => {
  it("renders", () => {
    expect.assertions(1);
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
