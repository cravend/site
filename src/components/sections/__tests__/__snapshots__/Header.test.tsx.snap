// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders 1`] = `
<div>
  <header
    class="diagonal"
  >
    <div
      class="container"
    >
      <div
        class="profile"
      >
        <img
          alt="Profile picture"
          class="pfp"
          data-nimg="1"
          decoding="async"
          height="150"
          loading="lazy"
          src="/_next/image?url=%2Fhello-world.jpg&w=384&q=75"
          srcset="/_next/image?url=%2Fhello-world.jpg&w=256&q=75 1x, /_next/image?url=%2Fhello-world.jpg&w=384&q=75 2x"
          style="color: transparent;"
          width="150"
        />
      </div>
      <div
        class="text_container"
      >
        <h1
          class="title"
        >
          Dalton Craven
        </h1>
        <p
          class="subtitle"
        >
          <b>
            Student & Developer
          </b>
        </p>
      </div>
    </div>
  </header>
</div>
`;
